src/firefly/js/core/LayoutCntlr.js

```diff
@@ -43,6 +43,10 @@ export function reducer(state={}, action={}) {
             if( dropDown ) {
                 action.payload.dropDown.view = getSelView(state, dropDown);
             }
+            //LZ 11/22/17  BUG here, the smart merge combine two table columns to one, if the target have n1  columns
+            //and the source has n2 columns, where n1>n2, the results is that the target first n2 columns were replaced
+            //by the source columns, the n1-n2 columns remain unchanged.  Thus, even though new table only has n2 columns,
+            // the state stored n1 columns.  This caused the search issue.
```

Comment:
This is not a bug.  It's designed to take the changes, and merge it with the original.  The changes you made below in LcManager is how you tell this function that changes were made to that prop/state.  You should remove this comment.

Reply:
haha, this was the first time I found the issue.  I forgot to remove it.  I will remove it and then merge.